wasm/demo/src/main.js

```diff
@@ -72,3 +73,55 @@ snippets.addEventListener('change', updateSnippet);
 // Run once for demo (updateSnippet b/c the browser might try to keep the same
 // option selected for the `select`, but the textarea won't be updated)
 updateSnippet();
+
+const prompt = ">>>>> ";
+
+const term = new Terminal();
+term.open(document.getElementById('terminal'));
+term.write(prompt);
+
+function remove_non_ascii(str) {
+    if ((str===null) || (str===''))
+        return false;
+    else
+        str = str.toString();
+
+    return str.replace(/[^\x20-\x7E]/g, '');
+}
+
+function print_to_console(data) {
+    term.write(remove_non_ascii(data) + "\r\n");
+}
+
+var input = "";
+term.on("data", (data) => {
+  const code = data.charCodeAt(0);
+  if (code == 13) { // CR
+    if (input[input.length - 1] == ':') {
+        input += data
+        term.write("\r\n.....");
+    } else {
+        term.write("\r\n");
+        try {
+            rp.pyEval(input, {
+                stdout: print_to_console
+            });
+        } catch (err) {
+            if (err instanceof WebAssembly.RuntimeError) {
+                err = window.__RUSTPYTHON_ERROR || err;
+            }
+            print_to_console(err);
+        }
+        term.write(prompt);
```

Comment:
I can press backspace and delete the prompt when typing, so maybe do a check in the backspace case if the length of the remaining characters in the line is equal to the length of the ps1/ps2.